scripts/unassign-inactive-issues.js

```diff
@@ -0,0 +1,257 @@
+          );
+
+          for (const ref of prReferences) {
+            try {
+              const prDetails = await github.rest.pulls.get({
+                owner,
+                repo,
+                pull_number: ref.source.issue.number
+              });
+              linkedPRs.push(prDetails.data);
+            } catch (prFetchError) {
+              console.error(`Error fetching PR details:`, prFetchError);
+            }
+          }
+
+          // Check for PR references in the issue body
+          const prLinkRegex = /(?:close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s+#(\d+)/gi;
+          const bodyMatches = [...(issue.body || '').matchAll(prLinkRegex)];
+          
+          for (const match of bodyMatches) {
+            try {
+              const prNumber = match[1];
+              const prDetails = await github.rest.pulls.get({
+                owner,
+                repo,
+                pull_number: prNumber
+              });
+              linkedPRs.push(prDetails.data);
+            } catch (prFetchError) {
+              console.error(`Error fetching PR from body link:`, prFetchError);
+            }
+          }
+
+          const openPRs = linkedPRs.filter(pr => pr.state === 'open');
+          
+          if (openPRs.length > 0) {
+            console.log(`Issue #${issue.number} has open PRs:`, 
+              openPRs.map(pr => `#${pr.number} (${pr.state})`));
+            return true;
+          }
+
+          console.log(`No open linked PRs found for issue #${issue.number}`);
+          return false;
+        } catch (error) {
+          console.error(`Error checking PR links for issue #${issue.number}:`, error);
+          return false;
+        }
+      };
+
+      const lastActivity = new Date(issue.updated_at);
+      const now = new Date();
+      
+      if (now - lastActivity > inactivityPeriodInMinutes * 60 * 1000) {
+        const hasOpenPRs = await checkLinkedPRs();
+        
+        if (hasOpenPRs) {
+          console.log(`Skipping issue #${issue.number} as it has open PRs`);
+          continue;
+        }
+
+        console.log(`Processing inactive issue #${issue.number} with no open PRs`);
+
+        const inactiveAssignees = [];
+        const activeAssignees = [];
+
+        for (const assignee of assignees) {
+          if (assignee.site_admin || await checkUserMembership(owner, repo, assignee.login)) {
+            activeAssignees.push(assignee.login);
+            continue;
+          }
+          inactiveAssignees.push(assignee.login);
+        }
+
+        if (inactiveAssignees.length === 0) continue;
+
+        try {
+          await github.rest.issues.update({
+            owner,
+            repo,
+            issue_number: issue.number,
+            assignees: activeAssignees,
+          });
+
+          console.log(`Successfully unassigned user from issue #${issue.number}`);
+
+          const mentionList = inactiveAssignees.map(login => `@${login}`).join(', ');
```

Comment:
I appreciate you've considered the situation with more assignees, it occasionally happens!